samples/ConPTY/EchoCon/EchoCon/EchoCon.cpp

```diff
@@ -0,0 +1,207 @@
+            // Create ConPTY
+            hr = CreatePseudoConsole(consoleSize, hPipePTYIn, hPipePTYOut, 0, &hPC);
+
+            // Initialize the necessary startup info struct        
+            if (S_OK == InitializeStartupInfoAttachedToConPTY(&startupInfo, hPC))
+            {
+                // Launch ping to echo some text back via the pipe
+                hr = CreateProcess(
+                    NULL,                           // No module name - use Command Line
+                    szCommand,                      // Command Line
+                    NULL,                           // Process handle not inheritable
+                    NULL,                           // Thread handle not inheritable
+                    TRUE,                           // Inherit handles
+                    EXTENDED_STARTUPINFO_PRESENT,   // Creation flags
+                    NULL,                           // Use parent's environment block
+                    NULL,                           // Use parent's starting directory 
+                    &startupInfo.StartupInfo,       // Pointer to STARTUPINFO
+                    &piClient)                      // Pointer to PROCESS_INFORMATION
+                    ? S_OK
+                    : GetLastError();
+
+                if (S_OK == hr)
+                {
+                    // Wait until ping completes
+                    WaitForSingleObject(piClient.hThread, 10 * 1000);
+
+                    // Allow listening thread to catch-up with final output!
+                    Sleep(500);
+                }
+
+                // --- CLOSEDOWN ---
+                // Close ConPTY - this will terminate client process if running
+                ClosePseudoConsole(hPC);
+
+                // Now safe to clean-up client process info process & thread
+                CloseHandle(piClient.hThread);
+                CloseHandle(piClient.hProcess);
+
+                // Cleanup attribute list
+                if (startupInfo.lpAttributeList)
+                {
+                    DeleteProcThreadAttributeList(startupInfo.lpAttributeList);
+                    free(startupInfo.lpAttributeList);
+                }
+            }
+            else
+            {
+                _tprintf_s(L"Error: Failed to initialize StartupInfo attached to ConPTY [0x%x]", 
+                    GetLastError());
+            }
+        }
+        else
+        {
+            _tprintf_s(L"Error: Failed to create pipes");
+        }
+
+        // Clean-up the pipes
+        if (hPipePTYOut) CloseHandle(hPipePTYOut);
+        if (hPipePTYIn) CloseHandle(hPipePTYIn);
+        if (hPipeOut) CloseHandle(hPipeOut);
+        if (hPipeIn) CloseHandle(hPipeIn);
+
+        // Restore console to its original mode.
+        hr = SetConsoleMode(hConsole, consoleMode)
+            ? S_OK
+            : GetLastError();
+    }
+
+    return hr == S_OK;
+}
+
+void __cdecl PipeListener(LPVOID pipe)
+{
+    HANDLE hPipe = (HANDLE)(HANDLE*)pipe;
```

Comment:
double cast, don't do that. also maybe prefer `static_cast` or `reinterpret_cast`